eclipse-theia/theia
# Add support for newer authentication API matching VS Code 1.63.1

#### What it does

Fixes https://github.com/eclipse-theia/theia/issues/9345
- Stay compatible with newer and previous authentication API
-- Keep prev authentication API matching VS Code 1.53.0-next.ea1b3f27db
-- Add type files for newer authentication API and implement interfaces
-- Bridge API version gap in plugin context and with merged types

- Add support for 'onAuthenticationRequest' activation event
-- Allow dedicated trigger through 'ensureProvider' call
-- Trigger activation if session of provider is requested

**Note:** This PR is based on the work of @PhilippeVienne done in this PR: https://github.com/eclipse-theia/theia/pull/9498

#### How to test
I created two authentication provider plugins based on [the sample app](https://github.com/clovis-dugue/theia-authenticationapi-pr) [mentioned](https://github.com/eclipse-theia/theia/pull/9498#issuecomment-869734727) in the previous PR: https://github.com/martin-fleck-at/theia/tree/issue-9345-test

- One plugin is using the older VS Code API
- One plugin is using the newer VS Code API
- Provide commands to log in and log out of an account, we can see notifications and the proper Theia UI integration

![authentication](https://user-images.githubusercontent.com/19170971/152337122-899da85a-0937-4491-a1ee-26d4dfcf34e6.gif)

- To test the `onAuthenticationRequest` activation, add `github-authentication` plugin

![authentication-github](https://user-images.githubusercontent.com/19170971/152337522-e34e710b-ff30-4f16-a66b-11e987ca67cc.gif)


#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
